### PR TITLE
Bug 1227219 - Try to get reproductible list of files

### PR DESCRIPTION
**Pull request by julienw**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ab60032d54683b2a8fd7cc4320ad4037f68ef4b9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=028f6cd290832ec012b6eef5b7ea86dd8cfa42ea

File: bin/gaia-marionette/gaia-marionette [hunk 1]
```diff
@@ -49,47 +52,24 @@ HARNESS_PATHS="$(cat "$BINDIR/../tests/jsmarionette/marionette_tests.txt")"
...
-                              -or -path "*/tv_apps/*/test/marionette/*_test.js" \
-                              -or -path "*/tests/integration/*/*_test.js")
-      else
-        TEST_FILES=""
-      fi
-
-      for wildcard in $HARNESS_PATHS; do
-        HARNESS_FILES=$(find "$BINDIR/../tests/jsmarionette" -path $wildcard)
-        TEST_FILES="$TEST_FILES $HARNESS_FILES"
-      done
```
Comment (julienw) [inline bin/gaia-marionette/gaia-marionette]: I also simplified the logic here; after all, if `$SKIP_TEST_FILES` is empty, there is no harm to use it, right ? so I removed this part of the if and kept only the second part.

I'm sure we can simplify further more (for example doing the find in a separate function) but I don't want to pack too much stuff here.

File: bin/gaia-marionette/gaia-marionette [hunk 2]
```diff
@@ -49,47 +52,24 @@ HARNESS_PATHS="$(cat "$BINDIR/../tests/jsmarionette/marionette_tests.txt")"
...
-      for wildcard in $HARNESS_PATHS; do
-        HARNESS_FILES=$(find "$BINDIR/../tests/jsmarionette" -path $wildcard $SKIP_TEST_FILES)
-        TEST_FILES="$TEST_FILES $HARNESS_FILES"
-      done
+      TEST_FILES=""
     fi
+
+    for wildcard in $HARNESS_PATHS; do
+      HARNESS_FILES=$(find "$BINDIR/../tests/jsmarionette" -path $wildcard $SKIP_TEST_FILES)
+      TEST_FILES="$TEST_FILES"$'\n'"$HARNESS_FILES" # $'\n' is a bashism to insert a litteral new line
```
Comment (julienw) [inline bin/gaia-marionette/gaia-marionette]: this is the real fix.

File: bin/gaia-marionette/gaia-marionette [hunk 3]
```diff
@@ -5,6 +5,9 @@
 # for more options.
 export MOZ_DISABLE_NONLOCAL_CONNECTIONS=1
 
+# Force the locale in this script so that it's reproducible
+export LC_ALL=C
+
```
Comment (julienw) [inline bin/gaia-marionette/gaia-marionette]: this is part of the fix as well

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=50289fb8fbe6bce7cd59aae9ad350bba31f97248

**Pull request merged**